pkg/plugins/placementpolicy/core/policy.go

```diff
@@ -0,0 +1,170 @@
+package core
+
+import (
+	"github.com/Azure/placement-policy-scheduler-plugins/apis/v1alpha1"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/kubernetes/pkg/scheduler/framework"
+)
+
+type PolicyInfo struct {
+	Namespace     string
+	Name          string
+	Action        v1alpha1.Action
+	TargetSize    *intstr.IntOrString
+	matchedPods   sets.String
+	qualifiedPods sets.String
+	managedPods   sets.String
+	targetMet     bool
+}
+
+func newPolicyInfo(namespace string, name string, action v1alpha1.Action, targetSize *intstr.IntOrString) *PolicyInfo {
+	policy := &PolicyInfo{
+		Namespace:     namespace,
+		Name:          name,
+		Action:        action,
+		TargetSize:    targetSize,
+		matchedPods:   sets.NewString(),
+		qualifiedPods: sets.NewString(),
+		managedPods:   sets.NewString(),
+		targetMet:     false,
+	}
+	return policy
+}
+
+type PolicyInfos map[string]map[string]*PolicyInfo
+
+func NewPolicyInfos() PolicyInfos {
+	return make(PolicyInfos)
+}
+
+func (p *PolicyInfo) merge(existing *PolicyInfo) *PolicyInfo {
+	existing.Action = p.Action
+	existing.TargetSize = p.TargetSize
+	existing.targetMet = p.targetMet
+
+	tempMatched := sets.NewString()
+	if len(p.matchedPods) > 0 {
+		pods := p.matchedPods.List()
+		for _, pod := range pods {
+			tempMatched = tempMatched.Insert(pod)
+		}
+	}
+	existing.matchedPods = tempMatched
+
+	tempQualified := sets.NewString()
+	if len(p.qualifiedPods) > 0 {
+		pods := p.qualifiedPods.List()
+		for _, pod := range pods {
+			tempQualified = tempQualified.Insert(pod)
+		}
+	}
+	existing.qualifiedPods = tempQualified
+
+	tempManaged := sets.NewString()
+	if len(p.managedPods) > 0 {
+		pods := p.managedPods.List()
+		for _, pod := range pods {
+			tempManaged = tempManaged.Insert(pod)
+		}
+	}
+	existing.managedPods = tempManaged
+
+	return existing
+}
+
+func (p *PolicyInfo) addMatch(pod *corev1.Pod) error {
+	key, keyError := framework.GetPodKey(pod)
```

Comment:
```suggestion
	key, err := framework.GetPodKey(pod)
```

ditto (error var name) for the entire package